compiler/rustc_middle/src/ty/context.rs

```diff
@@ -854,8 +856,14 @@ pub struct GlobalCtxt<'tcx> {
 
     /// Stores memory for globals (statics/consts).
     pub(crate) alloc_map: Lock<interpret::AllocMap<'tcx>>,
-
     current_gcx: CurrentGcx,
+
+    pub macro_map: RwLock<
+        FxHashMap<
+            LocalExpnId,
+            (TokenStream, Span, Lrc<dyn TcxMacroExpander + sync::DynSync + sync::DynSend>),
+        >,
+    >,
```

Comment:
This should not be public, but instead have a set of functions that ensure it can only be added to. It's also not safe for incremental (if macro expansions adds entries here, and the macro expansion query is not invoked again, but the result loaded from cache, the entries here won't get re-added)